tests/solana_tests/primitives.rs

```diff
@@ -941,10 +948,14 @@ fn test_overflow_detect_unsigned() {
             let limit = BigUint::from(2_u32).pow(width);
 
             // Generate a random number within the the range [0, 2^N -1]
-            let first_operand_rand = rng.gen_biguint(width.into()).add(1_u32);
+            let mut first_operand_rand = rng.gen_biguint(width.into());
+
+            if first_operand_rand == BigUint::from(0_u32) {
+                first_operand_rand = BigUint::from(1_u32);
+            }
 
             // Calculate a number that when multiplied by first_operand_rand, the result will overflow N bits
-            let second_operand_rand = limit.div(&first_operand_rand).add(1_usize);
+            let second_operand_rand = limit.div(&first_operand_rand).add(2_usize);
```

Comment:
The problem you have here is  `limit / first_operand` is an integer division which will round down. In other words, if limit is 256 and first_operand is 162, then `256 / 162 = 1` and 1 * 162 does not overflow 256.

I don't know where 2 comes from. This needs a big fat comment in *any case*.

Reply:
Basically, making the second operand larger will make the result overflow